packages/tpetra/core/src/Tpetra_Details_reallocDualViewIfNeeded.hpp

```diff
@@ -100,22 +100,52 @@ reallocDualViewIfNeeded (Kokkos::DualView<ValueType*, DeviceType>& dv,
   }
   else if (curSize < newSize) { // too small; need to reallocate
     if (needFenceBeforeRealloc) {
-      execution_space().fence ();
+      // FENCE REVIEW - NOW PASSING
+      //   Testing: This code is exercised by unit tests.
+      //   GTX960:  Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+      //   White:   Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+      //   Plan:    Remove fence. But not sure yet why fence was placed.
+      //   Notes:
+
+      // execution_space().fence ();
     }
     dv = dual_view_type (); // free first, in order to save memory
     // If current size is 0, the DualView's Views likely lack a label.
     dv = dual_view_type (curSize == 0 ? newLabel : dv.d_view.label (), newSize);
-    execution_space().fence ();
+
+    // FENCE REVIEW - NOW PASSING
+    //   Testing: This code is exercised by unit tests.
+    //   GTX960:  Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+    //   White:   Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+    //   Plan:    Remove fence. But not sure yet why fence was placed.
+    //   Notes:
+
+    // execution_space().fence ();
```

Comment:
This fence is not necessary. Kokkos view allocation and view fill value functor put fence with execution space. It is safe to touch.

Reply:
Kept and added comments #7546